cardinal/config.go

```diff
@@ -33,39 +48,104 @@ var defaultConfig = WorldConfig{
 type RunMode string
 
 type WorldConfig struct {
-	RedisAddress              string  `config:"REDIS_ADDRESS"`
-	RedisPassword             string  `config:"REDIS_PASSWORD"`
-	CardinalNamespace         string  `config:"CARDINAL_NAMESPACE"`
-	CardinalMode              RunMode `config:"CARDINAL_MODE"`
-	BaseShardSequencerAddress string  `config:"BASE_SHARD_SEQUENCER_ADDRESS"`
-	BaseShardQueryAddress     string  `config:"BASE_SHARD_QUERY_ADDRESS"`
-	CardinalLogLevel          string  `config:"CARDINAL_LOG_LEVEL"`
-	StatsdAddress             string  `config:"STATSD_ADDRESS"`
-	TraceAddress              string  `config:"TRACE_ADDRESS"`
+	// Cardinal
+	CardinalMode      RunMode `config:"CARDINAL_MODE"`
+	CardinalNamespace string  `config:"CARDINAL_NAMESPACE"`
+	CardinalLogLevel  string  `config:"CARDINAL_LOG_LEVEL"`
+
+	// Redis
+	RedisAddress  string `config:"REDIS_ADDRESS"`
+	RedisPassword string `config:"REDIS_PASSWORD"`
+
+	// Shard networking
+	BaseShardSequencerAddress string `config:"BASE_SHARD_SEQUENCER_ADDRESS"`
+	BaseShardQueryAddress     string `config:"BASE_SHARD_QUERY_ADDRESS"`
+
+	// Telemetry
+	StatsdAddress string `config:"STATSD_ADDRESS"`
+	TraceAddress  string `config:"TRACE_ADDRESS"`
+}
+
+func LoadWorldConfig() WorldConfig {
+	cfg := defaultConfig
+
+	// Load config from environment variables
+	if err := config.FromEnv().To(&cfg); err != nil {
+		log.Fatal().Err(err).Msg("Failed to load config")
+	}
+
+	// Validate config
+	if err := cfg.Validate(); err != nil {
+		log.Fatal().Err(err).Msg("Invalid config")
+	}
+
+	// Set logger config
+	if err := cfg.setLogger(); err != nil {
+		log.Fatal().Err(err).Msg("Failed to set log level")
+	}
+
+	return cfg
 }
 
-func (w WorldConfig) Validate() error {
-	if w.CardinalMode != RunModeProd {
-		return nil
+func (w *WorldConfig) Validate() error {
+	// Validate Cardinal config
+	if w.CardinalMode != RunModeProd && w.CardinalMode != RunModeDev {
+		return eris.Errorf("CARDINAL_MODE must be either %q or %q", RunModeProd, RunModeDev)
+	}
+	if err := Namespace(w.CardinalNamespace).Validate(w.CardinalMode); err != nil {
+		return eris.Wrap(err, "CARDINAL_NAMESPACE is not a valid namespace")
+	}
```

Comment:
hm, i dont know if that's useful for debugging; i'd imagine i want to know what env var in question does it corresponds it.

Reply:
maybe both?

Reply:
I want to avoid it being too long/ambiguous. I feel like if you are a world engine core dev (that would care about what the property names are) the you can just look easily in config.go

This isn't the case for game developers who would need the actual environment variables name to know whats going on and they typically don't have the world engine core source code readily available